server/sonar-server/src/main/java/org/sonar/server/qualitygate/QualityGates.java

```diff
@@ -377,4 +394,22 @@ private void checkPermission(Long projectId, DbSession session) {
       throw new ForbiddenException("Insufficient privileges");
     }
   }
+
+  private static class MatchQualityGate implements Predicate<QualityGateConditionDto> {
+    private final int metricId;
+    @CheckForNull
+    private final Integer period;
+
+    public MatchQualityGate(int metricId, @Nullable Integer period) {
+      this.metricId = metricId;
+      this.period = period;
+    }
+
+    @Override
+    public boolean apply(@Nonnull QualityGateConditionDto input) {
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) Remove this "Nonnull" annotation to honor the overridden method's contract. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://dory.sonarsource.com/sonarqube/coding_rules#rule_key=squid%3AS2638)